File: frontend/src/app/features/dashboard/requesters/requester.service.ts
import {Injectable} from '@angular/core';
import {AbstractCrudService} from '../../../core/services/abstract-crud.service';
import {RequesterResponseDto} from './dtos/requester.response.dto';
import {RequesterRequestDto} from './dtos/requester-request.dto';

@Injectable({providedIn: 'root'})
export class RequesterService extends AbstractCrudService<RequesterRequestDto, RequesterResponseDto> {
  constructor() {
    super('/requesters');
  }
}

File: backend/src/main/java/io/hb3nce04/keykeeperapp/service/RequesterService.java
package io.hb3nce04.keykeeperapp.service;

import org.springframework.stereotype.Service;

import io.hb3nce04.keykeeperapp.mapper.RequesterMapper;
import io.hb3nce04.keykeeperapp.model.dto.request.RequesterRequestDto;
import io.hb3nce04.keykeeperapp.model.dto.response.RequesterResponseDto;
import io.hb3nce04.keykeeperapp.model.entity.Requester;
import io.hb3nce04.keykeeperapp.repository.RequesterRepository;
import io.hb3nce04.keykeeperapp.service.common.AbstractCrudService;

@Service
public class RequesterService extends AbstractCrudService<Requester, RequesterRequestDto, RequesterResponseDto, RequesterRepository, RequesterMapper> {
    public RequesterService(
            RequesterRepository repository,
            RequesterMapper mapper) {
        super(repository, mapper);
    }
}
